### PR TITLE
Refactor/1001 disabled input

### PR DESCRIPTION
**Pull request by caseybaggz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
contributes #1001 

## What is the new behavior?
- Updates all Input-related APIs to use new shared styles & disabled selector
- Fixes bug in composes generation that removes the (now) unnecessary `extract` helper
- Adds some JS APIs to the sandbox env but not for all due to time

### CODE REVIEW

File: packages/headless-styles/src/components/Popover/generated/popoverCSS.module.ts [hunk 1]
```diff
@@ -39,7 +37,6 @@ export default {
     lineHeight: '1.25',
     padding: '1rem 2.5rem 1rem 1rem',
     '&::after': {
-      ...extract(tooltipCSS.tooltipContentBase, '&::after'),
```
Comment (Rykus0) [inline packages/headless-styles/src/components/Popover/generated/popoverCSS.module.ts:42]: In this situation, we need `extract` to merge the nested properties, so I don't think we can remove it (unless we handle the merge in the JS API)

File: packages/headless-styles/sandbox/plugins/css-to-ts/convert-inputs/write-file.js [hunk 2]
```diff
@@ -127,17 +127,22 @@ function getExternalComposeFileAndValue(composes) {
 
 function getExternalEntry(externalCompose, importMap) {
   const { valueName, file } = externalCompose
-  const moduleReg = /(\.?\.\/)+(.+)\/(.+)\.module\.css/
+  // const moduleReg = /(\.?\.\/)+(.+)\/(.+)\.module\.css/
```
Comment (Rykus0) [inline packages/headless-styles/sandbox/plugins/css-to-ts/convert-inputs/write-file.js:130]: We can probably remove these comments, right?

Comment (caseybaggz) [reply packages/headless-styles/sandbox/plugins/css-to-ts/convert-inputs/write-file.js:130]: Oh, totally.

**Comment by Rykus0** [comment]
Still working my way through, but it looks like there's an issue with the pagination buttons and JS tooltip (positioning)

image

image

File: packages/headless-styles/src/components/shared/helpers/tooltipHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  options: Required<TooltipOptions> | Required<PopoverOptions>
+): TooltipStyleKeys<StyleModule> {
+  return {
+    contentPositionClass:
+      `${options.position}PandoPositionContent` as StyleKey<StyleModule>,
+    positionClass: `${options.position}PandoPosition` as StyleKey<StyleModule>,
+  }
+}
+
+export function getTooltipPositions(position: Position) {
```
Comment (Rykus0) [inline packages/headless-styles/src/components/shared/helpers/tooltipHelpers.ts:16]: With `composes` working in the JS output now, we should be able to drop these helpers and reference the styles using the same classes as the CSS API.

Comment (caseybaggz) [reply packages/headless-styles/src/components/shared/helpers/tooltipHelpers.ts:16]: > Still working my way through, but it looks like there's an issue with the pagination buttons and JS tooltip (positioning)
> 
> image
> 
> image

Ah, shoot! I 100% made a mental note to update that and didn't! Good catch!

Comment (caseybaggz) [reply packages/headless-styles/src/components/shared/helpers/tooltipHelpers.ts:16]: > With `composes` working in the JS output now, we should be able to drop these helpers and reference the styles using the same classes as the CSS API.

Do you mind adding that in? As the author, you can probably knock that out faster async wise?

Comment (caseybaggz) [reply packages/headless-styles/src/components/shared/helpers/tooltipHelpers.ts:16]: Fixed pagination:
Screenshot 2023-02-01 at 1 30 17 PM

Comment (Rykus0) [reply packages/headless-styles/src/components/shared/helpers/tooltipHelpers.ts:16]: > Do you mind adding that in? As the author, you can probably knock that out faster async wise?

Yeah, I can do that.  Sorry I missed this earlier.  I'll hop on it now.

**Review by Rykus0 — APPROVED** [review]

**Pull request merged**